### ISSUE TITLE
wrong count for 'no-multiple-empty-lines' on last line

### ISSUE THREAD

**Issue by le0nik** [eslint/eslint#4228]
Eslint v1.7.3
In `.eslintrc`

Happens on the last line.

```
 "no-multiple-empty-lines": [2, {"max": 1}],
```

throws error with this code:

```
1 console.log(a);
2 <---- (error, although only 1 empty line)
```

```
2:1  error  Multiple blank lines not allowed  no-multiple-empty-lines
```

-------------------------------------------------

this rule
```
 "no-multiple-empty-lines": [2, {"max": 2}],
```

doesn't throw for this code

```
1 console.log(a);
2 
```

but does for this one
```
1 console.log(a);
2 
3 <----(error, although only two empty lines)
```

```
3:1  error  Multiple blank lines not allowed  no-multiple-empty-lines
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Cannot reproduce it on online demo. have you tried it [here](http://eslint.org/demo/)?

**Comment by nzakas** [comment]
@alberto

**Comment by alberto** [comment]
I'll look at this

**Comment by le0nik** [comment]
@gyandeeps can't reproduce it in online demo. One can't choose the number of max lines there, so i don't know whether it works correctly or not.

**Comment by alberto** [comment]
Ok, it turns out I didn't know what is a linebreak. :)

A linebreak doesn't really add a new line, just terminates the current line.

This code has just one line of code, linebreak terminated:
```js
var a = 5;\n
```
As shown by `wc` output:
```bash
wc -l linebreak.js
       1 linebreak.js
```

And the file output displaying non-printing characters
```
cat -e linebreak.js
var a = 5;$
```
However `context.getSourceLines()` returns 2 lines:
```js
console.log(context.getSourceLines())
[ 'var a = 5;', '' ]
```

And editors display that file differently:

Some, like Atom and Sublime Text 3 pretend there is an empty line for you to keep writing, when there is a linebreak terminating the line.
![atom](https://cloud.githubusercontent.com/assets/72561/10693120/a2c6458c-7996-11e5-9e02-7706411c5206.png)

I believe the same thing is happening in the [online editor](http://eslint.org/demo/)

Others, like vim don't
![vim](https://cloud.githubusercontent.com/assets/72561/10693600/8ff7146a-7999-11e5-83dc-22a47f057360.png)

Given all that, I'm not sure what should be done. Now I'm thinking [this fix](https://github.com/alberto/eslint/commit/617f41e319e2cc91a6f6f2af0b90012f979f55ee#diff-0f6472c36e2e4d88a28a92f9b5148206L58) was wrong, although the reporting is confusing in editors such as atom.

Thoughts @eslint/eslint-team ?

**Comment by nzakas** [comment]
Can you explain why you think that fix was wrong?

**Comment by alberto** [comment]
@nzakas It depends on wether you consider the example I showed has 0 or 1 empty lines at the end.

**Comment by alberto** [comment]
@gyandeeps what's your editor?

Can you do a `cat -e <file>`?

**Comment by le0nik** [comment]
@alberto i suppose, you wanted to ask me.

My editor is macvim with syntastic.

Here's the result of running `cat -e <file>` command:

```
console.log('hello');$
$
```

**Comment by alberto** [comment]
Yes @le0nik, thanks

**Comment by nzakas** [comment]
@alberto can you break it down for me? In what scenario was the fix wrong? Why was it wrong?

**Comment by alberto** [comment]
If you consider the linebreak to be a line terminator but not creating a new line itself, then I think the original code was doing the right thing removing the last line if it was empty.

As you can see from the screenshots I posted, when you open the same file with vim it appears to have one line. With atom it looks like it has two. I don't think there is a way for them to look correct from a user perspective, but technically the original behaviour seems the correct one so I would favor that.

**Comment by nzakas** [comment]
Thanks for explaining, I agree. Do you want to make that change?

**Comment by alberto** [comment]
yes, let me fix that
